dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallback.java

```diff
@@ -337,13 +340,27 @@ public void initWithHosts(Map<Host, HostConnectionPool<CL>> hPools) {
 		
 		Set<String> remoteDCs = new HashSet<String>();
 
+		/* Initialize the hashtag with the first host (if hashtag is defined) */
+		this.hashtag = hPools.keySet().iterator().next().getHashtag();
+		
 		for (Host host : hPools.keySet()) {
 			String dc = host.getRack();
 			if (localRack != null && !localRack.isEmpty() && dc != null && !dc.isEmpty() && !localRack.equals(dc)) {
 				remoteDCs.add(dc);
 			}
+			/**
+			 * Checking for defined hashtags if all of them are the same.
+			 * If not we need to throw an exception.
+			 */
+			if(this.hashtag != null) {
```

Comment:
We initialize hosts and we iterate over all the `Hosts` and make sure that all hashtags across all hosts are the same.

Reply:
negative test case for different hashtags

Reply:
check for empty hashtags